docs/source/rbac/scopes.md

```diff
@@ -0,0 +1,37 @@
+# Scopes
+
+A scope has a syntax-based design that reveals which resources it provides access to:
+- The base`resource` scopes, such as `users` or `groups`, provides non-elevated 'default' read or write permissions to the resource itself and all sub-resources. For instance, the scope`users:servers` is included within the scope`users`.
+- The elevated `admin:resource` scopes extend permissions beyond default. For instance, where the scope `users` provides read and write access to user information, the scope `admin:users` allows creating and deleting users.
+- The scope structure `resource:subresource` vertical filtering: it provides access to a subset of the information granted by the `resource` scope. For example, the scope `users:names` provides access to user names only.
+- The scope structure `read:resource` (or `read:resource:subresource`) limits permissions to read-only operations on `resource` (or `subresource`).
+- The scope structure `resource!user=charlie` allows for horizontal filtering: it limits access to only return resources of user `charlie`. Only one filter per scope is allowed, but filters for the same scope have an additive effect; a larger filter can be used by supplying the scope multiple times with different filters.
```

Comment:
When I first skimmed though the docs I missed this explanation of what `!` means as the text is quite dense:
![image](https://user-images.githubusercontent.com/1644105/108517438-54aee480-72bf-11eb-852c-05f831917c15.png)

Since it's introducing new syntax compared to the previous page is there a way to highlight this? A use-case I'm thinking of is someone reading a `jupyterhub_config.py` sees some scopes for the first time. The `aaa:bbb` format is similar to that used in many other projects, but the `!` is less familiar.

Might also be nice to cross-reference "horizontal filtering" or provide a brief in-line explanation since it's new terminology that (at least to me) doesn't have an intuitive meaning.

Reply:
The whole section is now reorganized to highlight the different scope syntax with added cross-referencing of filtering sections.